### DISCUSSION TITLE
mdx to md converter?

### DISCUSSION THREAD

**Discussion by deadcoder0904** [unifiedjs/unified#184] [Q&A] [answered]
i'm looking to generate pdf of one of my blog posts & it seems like `pandoc` which i plan to use needs `md` to convert it to `pdf` so i was wondering if someone has already built `mdx` to `md` converter?

i think the only change i would need is converting an `img` tag that is used as an `import` to the markdown `img` syntax.

so how do i do it?

additionally, if i add custom components in `mdx` what should i use for converting that?

**Comment by ChristianMurphy** [comment, marked as answer]
The short answer, you can render the HTML output from MDX directly to PDF (using Chrome, or a wrapper like [Playwright](https://playwright.dev/docs/cli#generate-pdf) or [Puppeteer](https://github.com/puppeteer/puppeteer/blob/v13.3.2/docs/api.md#pagepdfoptions)), no need to go through pandoc first.

If you really want to go through pandoc, writing an MDX converter still probably doesn't make sense.

A few examples to illustrate why:
1. Pandoc can render HTML to PDF, and could be used as an alternative to the browser based converting solutions noted above. Though it may lose more of the intended formatting than a browser would.
2. If you use MDX for markdown without any JSX, in this case content already is valid markdown, no converter is needed
3. If you use MDX for markdown with intrinsic JSX (html tags) only, in this case content is already valid markdown, no converter is needed
4. If you use MDX with JSX and custom components, in this case you cannot convert the custom components without resolving them using React, Vue, etc. You probably want to use [MDX evaluate](https://mdxjs.com/packages/mdx/#evaluatefile-options) with [Render to Static Markup](https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup) and pass that content though [rehype-remark](https://github.com/rehypejs/rehype-remark) to take that static HTML back to static markdown. Though this would lose even more of the intended formatting than option 1 above.

**Reply by deadcoder0904** [reply]
that answers all my questions, thanks christian :)

**Reply by deadcoder0904** [reply]
[this playwright script](https://www.checklyhq.com/learn/headless/generating-pdfs/) generates pdf that look good enough so i went with that :)
